pkg/tools/dnsutils/fanout/handler.go

```diff
@@ -55,6 +56,11 @@ func (h *fanoutHandler) ServeDNS(ctx context.Context, rw dns.ResponseWriter, msg
 				Timeout: timeout,
 			}
 
+			// If u.Host is IPv6 then wrap it in brackets
+			if strings.Count(u.Host, ":") >= 2 && !strings.HasPrefix(u.Host, "[") && !strings.Contains(u.Host, "]") {
+				u.Host = fmt.Sprintf("[%s]", u.Host)
```

Comment:
Can we apply this for`address` variable from line 64?

Reply:
Unfortunately no. We need line 65.
`url.Port()` validates the value when calculating the port. And it looks at brackets.
So we need to modify `u.Host`